src/modular/residue.rs

```diff
@@ -39,6 +39,8 @@ pub trait ResidueParams<const LIMBS: usize>:
 
     /// The constant modulus
     const MODULUS: Uint<LIMBS>;
+    /// The constant modulus pre-wrapped in NonZero
+    const MODULUS_NZ: NonZero<Uint<LIMBS>>;
```

Comment:
Perhaps the modulus should just be `NonZero` to begin with?

Reply:
Good idea. Fixed. Also note that I removed `Residue::new_checked()` and the check in `Residue::new()` because `ResidueParams` is already guaranteed to have an odd modulus.

Reply:
Also had to add a `const fn as_ref()` to `NonZero` to use in macros